storage/pruning/pruningStorer.go

```diff
@@ -967,8 +984,10 @@ func (ps *PruningStorer) processPersistersToClose() []*persisterData {
 	}
 
 	if ps.numOfActivePersisters < uint32(len(ps.activePersisters)) {
-		persistersToClose = ps.activePersisters[ps.numOfActivePersisters:]
-		ps.activePersisters = ps.activePersisters[:ps.numOfActivePersisters]
+		numPersistersToKeep := ps.getNumPersistersToKeep(lastEpochNeeded)
+		epochRemoveIndex := core.MaxUint32(numPersistersToKeep, ps.numOfActivePersisters)
```

Comment:
I would have called this epochRemoveIndex allowedNumPersistersToKeep as the index naming doesn't tell too much

Reply:
Done.